modules/tinymce/src/core/main/ts/init/InitContentBody.ts

```diff
@@ -50,8 +51,95 @@ const appendStyle = function (editor: Editor, text: string) {
   Insert.append(head, tag);
 };
 
+const getRootName = (editor: Editor): string => editor.inline ? editor.getElement().nodeName.toLowerCase() : undefined;
+
+const removeUndefined = <T>(obj: T): T => {
+  return Obj.filter(obj as Record<string, unknown>, (v) => Type.isUndefined(v) === false) as T;
+};
+
+const mkParserSettings = (editor: Editor): DomParserSettings => {
+  const settings = editor.settings;
+  const blobCache = editor.editorUpload.blobCache;
+
+  return removeUndefined<DomParserSettings>({
+    allow_conditional_comments: settings.allow_conditional_comments,
+    allow_html_in_named_anchor: settings.allow_html_in_named_anchor,
+    allow_script_urls: settings.allow_script_urls,
+    allow_unsafe_link_target: settings.allow_unsafe_link_target,
+    convert_fonts_to_spans: settings.convert_fonts_to_spans,
+    fix_list_elements: settings.fix_list_elements,
+    font_size_legacy_values: settings.font_size_legacy_values,
+    forced_root_block: settings.forced_root_block,
+    forced_root_block_attrs: settings.forced_root_block_attrs,
+    padd_empty_with_br: settings.padd_empty_with_br,
+    preserve_cdata: settings.preserve_cdata,
+    remove_trailing_brs: settings.remove_trailing_brs,
+    inline_styles: settings.inline_styles,
+    root_name: getRootName(editor),
+    validate: true,
+    blob_cache: blobCache
+  });
+};
+
+const mkSerializerSettings = (editor: Editor): SerializerSettings => {
+  const settings = editor.settings;
+
+  return removeUndefined<SerializerSettings>({
```

Comment:
Can this use the `mkParserSettings` to remove duplication, as serializer settings extend parser settings?

Reply:
I guess we could run that and merge the two objects. Saves a few bytes of code.